common/src/abstractions/folder.service.ts

```diff
@@ -1,21 +1,19 @@
 import { FolderData } from "../models/data/folderData";
 import { Folder } from "../models/domain/folder";
-import { SymmetricCryptoKey } from "../models/domain/symmetricCryptoKey";
 import { TreeNode } from "../models/domain/treeNode";
 import { FolderView } from "../models/view/folderView";
 
 export abstract class FolderService {
   clearCache: (userId?: string) => Promise<void>;
-  encrypt: (model: FolderView, key?: SymmetricCryptoKey) => Promise<Folder>;
   get: (id: string) => Promise<Folder>;
   getAll: () => Promise<Folder[]>;
   getAllDecrypted: () => Promise<FolderView[]>;
   getAllNested: () => Promise<TreeNode<FolderView>[]>;
   getNested: (id: string) => Promise<TreeNode<FolderView>>;
   saveWithServer: (folder: Folder) => Promise<any>;
-  upsert: (folder: FolderData | FolderData[]) => Promise<any>;
+  upsert: (folder: FolderData) => Promise<any>;
```

Comment:
Are you planning on an `upsertMany` if a bulk operation is supported?

Reply:
Yes, I think that's a cleaner pattern than allowing two different data models.

`upsert(folder) => upsertMany([folder])`